### ISSUE TITLE
[Question] Testing controllers with RSpec

### ISSUE THREAD

**Issue by lucasmedeirosleite** [waiting-for-dev/devise-jwt#59]
Hi!

I'm building a Rails 5 app (in `--api` mode) and I would like to test my controllers,
I tried what the devise wiki tell us to do:

```ruby
@request.env["devise.mapping"] = Devise.mappings[:user]
sign_in FactoryGirl.create(:user)
```

But after I perform a post I still receives `401` and the actions is not processed. :(

**Comment by waiting-for-dev** [comment]
Hi!

That helper stores the user in the session, but Rails `--api` has the session disabled. However, even if you enable the session it won't work because of https://github.com/waiting-for-dev/warden-jwt_auth/issues/12.

You should provide the token manually at each request in the request headers. You can get the token from a sign_in or you can create it manually with [UserEncoder](https://github.com/waiting-for-dev/warden-jwt_auth/blob/master/lib/warden/jwt_auth/user_encoder.rb).

Maybe it is worth to comment it in the README or maybe to create some kind of test helper. I'll study it.

**Comment by waiting-for-dev** [comment]
Ok, added a test helper to authenticate request headers for a user. You can read how to use it in the [README](https://github.com/waiting-for-dev/devise-jwt#testing).

Released in version `0.5.2`

**Comment by lucasmedeirosleite** [comment]
Thanks!
